### PR TITLE
notifier: Moved config package dependency from notifier package.

### PR DESCRIPTION
**Pull request by bwplotka**
Follow up from https://github.com/prometheus/prometheus/pull/4955 to make notifier independent of config package.

~TODO in next PR: Move notifier to `pkg/notifier`, @bbrazil what do you think?~

Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

**Comment by brian-brazil** [comment]
This looks okay from a first glance. I don't see a reason we'd move this to pkg, it's not a utility library.

**Comment by bwplotka** [comment]
Hm.. I thought it is a golang convention to have all in `pkg` and thought we are moving towards it. Ignore me then, removed this todo from description.

File: config/config.go [hunk 1]
```diff
@@ -456,10 +445,10 @@ func (c *AlertmanagerConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
 		return err
 	}
 
-	// The UnmarshalYAML method of HTTPClientConfig is not being called because it's not a pointer.
+	// The UnmarshalYAML method of NotifierConfig.HTTPClientConfig is not being called because it's not a pointer.
 	// We cannot make it a pointer as the parser panics for inlined pointer structs.
 	// Thus we just do its validation here.
-	if err := c.HTTPClientConfig.Validate(); err != nil {
+	if err := c.NotifierConfig.HTTPClientConfig.Validate(); err != nil {
```
Comment (brian-brazil) [inline config/config.go]: Shouldn't this check move down?

File: notifier/notifier.go [hunk 2]
```diff
@@ -553,7 +564,7 @@ type alertmanagerSet struct {
 	logger     log.Logger
 }
 
-func newAlertmanagerSet(cfg *config.AlertmanagerConfig, logger log.Logger) (*alertmanagerSet, error) {
+func newAlertmanagerSet(cfg Config, logger log.Logger) (*alertmanagerSet, error) {
```
Comment (brian-brazil) [inline notifier/notifier.go]: You're passing around full structs rather than pointers, just pointers would be more efficient.

**Comment by bwplotka** [comment]
PTAL @brian-brazil  (:  This is last (known) piece that stops us from upgrading Prometheus dependency in Thanos codebase.

**Comment by brian-brazil** [comment]
I was looking at it honest, but paused to take a shower :)

File: config/config.go (hunk 1 above)
Comment (brian-brazil) [reply config/config.go]: I meant moved down to NotifierConfig, due to Law of Demeter

**Comment by bwplotka** [comment]
Definitely stopped doing this. Help wanted to continue this :hugs: 

Let's also update an issue with latest for removing config deps or rather some idea of indirection for config.

**Comment by bwplotka** [comment]
Actually this is no longer needed if https://github.com/prometheus/prometheus/issues/7107 will be done.

**Pull request closed without merging**